### PR TITLE
Jenkinsfile: skip CI for changes to README

### PR DESCRIPTION
**Pull request by nikhil-zinjurde-imgtec**
Signed-off-by: Nikhil Zinjurde <nikhil.zinjurde@imgtec.com>

### CODE REVIEW

**Comment by Shpinkso** [comment]
Does this connect to openwrt/#229 ? Why is this being done before openwrt? CI usage is neglidgable for uboot

File: Jenkinsfile [hunk 1]
```diff
@@ -15,6 +24,18 @@ node('docker && imgtec') {
                     [$class: 'PruneStaleBranch'],
                 ],
             ])
+            // Skip CI if changes in README, else continue building
+            sh 'git show --pretty="format:" --name-only --no-merges 89ffc511f185058ac6bf3fdaa44c932227d3416e > commandResult'
```
Comment (Ham22) [inline Jenkinsfile:28]: All of this information should already be in the jenkins scm object as it is able to display a list of changes files on the web page.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,3 +1,12 @@
+#!/usr/bin/env groovy
+
+def excludePatterns = [
```
Comment (Ham22) [inline Jenkinsfile:3]: I would expect patterns to work with either regex or glob not static strings as currently `.md` is very ambiguous.

File: Jenkinsfile [hunk 3]
```diff
@@ -15,6 +24,18 @@ node('docker && imgtec') {
...
                 ],
             ])
+            // Skip CI if changes in README, else continue building
+            sh 'git show --pretty="format:" --name-only --no-merges 89ffc511f185058ac6bf3fdaa44c932227d3416e > commandResult'
+            String[] result = new File('commandResult')
+            echo "Git commit file(s): ${result}"
+            echo "First element: ${result[0]}"
+            for (pattern in excludePatterns) {
+                if (result.size() == 1 && result[0] ==~ /.*$${pattern}/) {
+                    echo 'Changes in README, skipping CI'
```
Comment (Ham22) [inline Jenkinsfile:34]: No longer true

File: Jenkinsfile [hunk 4]
```diff
@@ -15,6 +24,18 @@ node('docker && imgtec') {
                     [$class: 'PruneStaleBranch'],
                 ],
             ])
+            // Skip CI if changes in README, else continue building
+            sh 'git show --pretty="format:" --name-only --no-merges 89ffc511f185058ac6bf3fdaa44c932227d3416e > commandResult'
+            String[] result = new File('commandResult')
+            echo "Git commit file(s): ${result}"
+            echo "First element: ${result[0]}"
+            for (pattern in excludePatterns) {
+                if (result.size() == 1 && result[0] ==~ /.*$${pattern}/) {
```
Comment (Ham22) [inline Jenkinsfile:33]: This logic would only capture a subset right? Why only allowing there to be a single result? I would expect the code to skip a full build for the below scenario:

commit 4 - foo/Makefile
commit 3 - changed LICENCE
commit 2 - changed foo/bar/test.md
commit 1 - changed README.md

**Review by Ham22 — CHANGES_REQUESTED** [review]

**Comment by nikhil-zinjurde-imgtec** [comment]
Closing this and will come up with a different approach and for a different repo as recommended in the review comments

**Pull request closed without merging**